### PR TITLE
Add pprof heap endpoints to node api

### PR DESCRIPTION
**Pull request by jackkleeman**
Provide a mechanism of obtaining pprof-format heap dumps

On linux (easiest on musl build ):
```
curl localhost:5122/debug/pprof/heap/activate
... do some work that will lead to allocations
curl localhost:5122/debug/pprof/heap > heap.pb.gz
pprof -http=:9090 ./restate-server  heap.pb.gz
```

### CODE REVIEW

**Comment by jackkleeman** [comment]
[profile001.pdf](https://github.com/user-attachments/files/18062284/profile001.pdf)

File: crates/node/src/network_server/service.rs [hunk 1]
```diff
@@ -44,6 +45,9 @@ impl NetworkServer {
         // -- HTTP service (for prometheus et al.)
         let axum_router = axum::Router::new()
             .route("/metrics", get(render_metrics))
+            .route("/debug/pprof/heap", get(pprof::heap))
+            .route("/debug/pprof/heap/activate", get(pprof::activate_heap))
+            .route("/debug/pprof/heap/deactivate", get(pprof::deactivate_heap))
```
Comment (AhmedSoliman) [inline crates/node/src/network_server/service.rs:50]: Should we make those as POST?

File: crates/node/src/network_server/service.rs [hunk 2]
```diff
@@ -44,6 +45,9 @@ impl NetworkServer {
         // -- HTTP service (for prometheus et al.)
         let axum_router = axum::Router::new()
             .route("/metrics", get(render_metrics))
+            .route("/debug/pprof/heap", get(pprof::heap))
```
Comment (AhmedSoliman) [inline crates/node/src/network_server/service.rs:48]: perhaps?
```suggestion
            .route("/debug/pprof/dump_heap", get(pprof::heap))
```

**Review by AhmedSoliman — APPROVED** [review]
Very very nice. I've left a couple of tiny nits but I'm not sure if they're worth doing, will leave it up to you to decide. I'm good either way.

File: crates/node/src/network_server/service.rs (hunk 2 above)
Comment (jackkleeman) [reply crates/node/src/network_server/service.rs:48]: /heap is pretty common as its what go uses by default, so would probably keep this one

File: crates/node/src/network_server/service.rs (hunk 1 above)
Comment (jackkleeman) [reply crates/node/src/network_server/service.rs:50]: activate and deactivate yes it should probably be PUT

Comment (AhmedSoliman) [reply crates/node/src/network_server/service.rs:50]: PUT will be _very_ easy to forget :)

Comment (jackkleeman) [reply crates/node/src/network_server/service.rs:50]: one wonders if we should just accept any of them then :D

**Pull request merged**